src/Microsoft.ML.Data/Evaluators/Metrics/ConfusionMatrix.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.Data
+{
+    /// <summary>
+    /// Represents the <a href="https://en.wikipedia.org/wiki/Confusion_matrix">confusion matrix</a> of the classification results.
+    /// </summary>
+    public sealed class ConfusionMatrix
+    {
+        /// <summary>
+        /// The calculated value of <a href="https://en.wikipedia.org/wiki/Precision_and_recall#Precision">precision</a> for each class.
+        /// </summary>
+        public IReadOnlyList<double> PerClassPrecision { get; }
+
+        /// <summary>
+        /// The calculated value of <a href="https://en.wikipedia.org/wiki/Precision_and_recall#Recall">recall</a> for each class.
+        /// </summary>
+        public IReadOnlyList<double> PerClassRecall { get; }
+
+        /// <summary>
+        /// The confsion matrix counts for the combinations actual class/predicted class.
+        /// The actual classes are in the rows of the table, and the predicted classes in the columns.
+        /// </summary>
+        public IReadOnlyList<IReadOnlyList<double>> Counts { get; }
+
+        /// <summary>
+        /// The indicators of the predicted classes.
+        /// It might be the classes names, or just indices of the predicted classes, if the name mapping is missing.
+        /// </summary>
+        public int NumberOfPredictedClasses { get; }
```

Comment:
Is this just meant to give the  number of classes?
Maybe we could simplify both the name and the comment.

Suggestion: `NumberOfClasses`